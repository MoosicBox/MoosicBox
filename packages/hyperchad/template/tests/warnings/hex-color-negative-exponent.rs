//! Test case for hex colors with negative exponent pattern
//! 
//! Pattern like #1e-2 contains a minus sign, which breaks tokenization.
//! Rust's lexer would tokenize this as multiple tokens: # + 1e + - + 2
//! The minus sign makes this invalid as a hex color.
//!
//! This is not a valid hex color pattern in any case.

use hyperchad_template::container;

fn main() {
    // This will fail - minus sign is not a valid hex digit
    let _ = container! {
        div color=#1e-2 { "Negative exponent with minus sign" }
    };
}
