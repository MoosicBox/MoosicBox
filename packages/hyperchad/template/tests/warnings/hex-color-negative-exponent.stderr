error: Expected hex digits after '#' for hex color
  --> tests/warnings/hex-color-negative-exponent.rs:14:19
   |
14 |         div color=#1e-2 { "Negative exponent with minus sign" }
   |                   ^
   |
   = help: Note: Hex colors ending in 'e' without digits after (like #12e) must use string syntax: color="#12e"
   = help: This is due to Rust treating 'e' as scientific notation (e.g., 1e2 = 100.0)

error: expected one of: identifier, integer literal, string literal, `-`, `:`
  --> tests/warnings/hex-color-negative-exponent.rs:14:20
   |
14 |         div color=#1e-2 { "Negative exponent with minus sign" }
   |                    ^^^^
